### PR TITLE
FAD-5595: Basic filtering of message events

### PR DESCRIPTION
**Pull request by rajumsys**
Should be straightforward, no special things needed to start testing. 

For recipients field, i tried to make it smarter and catch common typing issues like comma, whitespaces etc. so values like `email@domain.com , email2@domain.com` should be fine. let me know if any other things should be caught. values with multiple commas won't work but i thought it's not normal :)

### CODE REVIEW

**Comment by rajumsys** [comment]
@bkemper i've updated with pretty decent test coverage.

File: src/pages/reports/messageEvents/MessageEventsPage.js [hunk 1]
```diff
@@ -55,32 +62,49 @@ export class MessageEventsPage extends Component {
...
       ? <Empty message={emptyMesasage} />
       : (
-        <TableCollection
-          columns={columns}
-          rows={events}
-          getRowData={this.getRowData}
-          pagination={true}
-        />
+        <div>
+          { events.length === maxResults &&
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/MessageEventsPage.js]: If we encounter an off-by-one issue in future, this test is a little brittle. Maybe `>=` ?

File: src/pages/reports/messageEvents/components/BasicFilter.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        relativeRange: relativeRange
+      }
+    }, this.refresh);
+  }
+
+  emailValidator = (value) => {
+    const invalids = this.getInvalidAddresses(this.parseAddresses(value));
+
+    if (invalids.length) {
+      return `${invalids.join(', ')} ${invalids.length > 1 ? 'are' : 'is'} not valid email ${invalids.length > 1 ? 'adresses' : 'address'}`;
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/BasicFilter.js]: `s/adresses/addresses/`

If we're going down the path of singular/plural detection, lets make sure we say "not **a** valid email address" in the singular case. I wonder if we could just say `Invalid: ...` though.

File: src/pages/reports/messageEvents/components/BasicFilter.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  render() {
+    return (
+      <Grid>
+        <Grid.Column xs={12} md={6}>
+          <DateFilter refresh={this.handleDateSelection} />
+        </Grid.Column>
+        <Grid.Column xs={12} md={6}>
+          <Field
+            name="recipients"
+            onBlur={this.handleRecipientsChange}
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/BasicFilter.js]: Can we avoid refreshing when we move focus away from the filter field when the field contents have not changed?

I also instinctively hit return after typing into search fields. Can we make that refresh the list?

File: src/pages/reports/messageEvents/components/BasicFilter.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            validate={this.emailValidator}
+            placeholder='recipient1@domain.com, recipient2@domain.com'
+          />
+        </Grid.Column>
+      </Grid>
+    );
+  }
+}
+
+const formName = 'basicFilter';
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/BasicFilter.js]: My understanding is the redux-form names are global. Maybe we should namespace this one a bit more. Something like `msgEventsBasicFilter` ?

File: src/pages/reports/messageEvents/components/tests/BasicFilter.test.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+  describe('getInvalidAddresses', () => {
+    it('detect invalid email addresses correctly', () => {
+      expect(instance.getInvalidAddresses(['email@domain.com', 'email2m'])).toEqual(['email2m']);
+      expect(instance.getInvalidAddresses(['abc'])).toEqual(['abc']);
+      expect(instance.getInvalidAddresses([''])).toEqual([]);
+      expect(instance.getInvalidAddresses(['email@domain.com'])).toEqual([]);
+    });
+  });
+
+  describe('emailValidator', () => {
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/tests/BasicFilter.test.js]: For reference, I have a set of validation helper tests in my support panel PR [here](https://github.com/SparkPost/2web2ui/pull/215/files#diff-9f4eb7044f3b220cde32ebaca62d16cb).

File: src/pages/reports/messageEvents/components/tests/BasicFilter.test.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+      expect(props.onSubmit).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('handleDateSelection', () => {
+    const now = new Date();
+
+    beforeEach(() => {
+      dateMock.getRelativeDates = jest.fn(() => {
+        return {
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/tests/BasicFilter.test.js]: Please `npm run lint`- you can remove the `return { }`: `jest.fn(() => ({ to: ..., from: ... }))`

File: src/pages/reports/messageEvents/components/BasicFilter.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+      reportFilters,
+      recipients: ''
+    };
+  }
+
+  refresh = () => {
+    this.props.onSubmit(this.state);
+  }
+
+  parseAddresses = (value) => {
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/BasicFilter.js]: Nice human-safe parsing

File: src/pages/reports/messageEvents/components/BasicFilter.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+          <DateFilter refresh={this.handleDateSelection} />
+        </Grid.Column>
+        <Grid.Column xs={12} md={6}>
+          <Field
+            name="recipients"
+            onBlur={this.handleRecipientsChange}
+            component={TextFieldWrapper}
+            title="Recipient Email(s)"
+            validate={this.emailValidator}
+            placeholder='recipient1@domain.com, recipient2@domain.com'
```
Comment (ewandennis) [inline src/pages/reports/messageEvents/components/BasicFilter.js]: Can we use an active voice call to action in the placeholder text? Something like `Filter by recipient email address`

File: src/pages/reports/messageEvents/MessageEventsPage.js (hunk 1 above)
Comment (rajumsys) [reply src/pages/reports/messageEvents/MessageEventsPage.js]: that would be breaking change for our API ;). i really wanted it to break in such case as
- api will return more than 1k and we're not limiting in frontend. so it'll say we're showing 1000 but actually showing more
- it's a api change.

File: src/pages/reports/messageEvents/components/BasicFilter.js (hunk 8 above)
Comment (ewandennis) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: Note sure how we make it clear we can enter lists of addresses there...

File: src/pages/reports/messageEvents/MessageEventsPage.js (hunk 1 above)
Comment (rajumsys) [reply src/pages/reports/messageEvents/MessageEventsPage.js]: alternative is to limit UI from showing max 1K and then changing to `>=` would make sense.

File: src/pages/reports/messageEvents/components/BasicFilter.js (hunk 2 above)
Comment (rajumsys) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: yeah, `not a ` sounds better.

File: src/pages/reports/messageEvents/components/BasicFilter.js (hunk 3 above)
Comment (rajumsys) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: Both are good points, I'll address that.

Comment (rajumsys) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: Seems MB's TextField doesn't support a onEnter (or even onKeyPress) event that'll let me detect when Return key is pressed. I guess, we'll have to implement that in MB and then update that in all filters in UI.

File: src/pages/reports/messageEvents/components/BasicFilter.js (hunk 8 above)
Comment (rajumsys) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: yeah, that's the reason I put the example and made them like `recipientN` so people know what email is this. i'm happy to change to something that means both things and format (comma separated)

File: src/pages/reports/messageEvents/components/BasicFilter.js (hunk 3 above)
Comment (rajumsys) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: Actually nvm, seems redux `Field` has support for `onKeyDown`. It'll work for this case.

File: src/pages/reports/messageEvents/MessageEventsPage.js (hunk 1 above)
Comment (ewandennis) [reply src/pages/reports/messageEvents/MessageEventsPage.js]: I meant we should change the test on the client side to make it resilient to something upstream giving us an array with an off-by-one number of results in it.

**Comment by bkemper** [comment]
@ewandennis It appears all your feedback addressed.  I took a quick look at the code, but didn't want to duplicate your work.  I focused more on smoke testing it before ask for the thumbs up to merge.  I notice this bug in the custom time range.  I'll have to dig in tomorrow.

message-event-filter

**Comment by ewandennis** [comment]
@bkemper Sounds good. A few quick points before we merge:

 - My point about the edge case in `src/pages/reports/messageEvents/MessageEventsPage.js` was not addressed. I don't think I was being clear enough. Could you please skim that note above and address?  
 - If the time range issue is app-wide, we can address separately
 - I'll do a last walk through then sign off after ☝️

**Comment by bkemper** [comment]
I wasn't able to reproduce that bug on UAT.  I'm triaging now to see what could be the cause.

File: src/pages/reports/messageEvents/MessageEventsPage.js (hunk 1 above)
Comment (bkemper) [reply src/pages/reports/messageEvents/MessageEventsPage.js]: Addressed... cf698e66

File: src/pages/reports/messageEvents/components/BasicFilter.js (hunk 8 above)
Comment (bkemper) [reply src/pages/reports/messageEvents/components/BasicFilter.js]: Addressed... 57474793

**Comment by bkemper** [comment]
@ewandennis I confirmed that the bug was due to user error.  I was entering a "to" time greater than the present time.  The filter only allows a time range in the past.  Here is the condition.

`if (to.isValid() && from.isValid() && from.isBefore(to) && to.isBefore(now))`

I submitted a follow-up ticket, [FAD-6003](https://jira.int.messagesystems.com/browse/FAD-6003), to improve this experience to avoid the same confusion I had.

With all that said, this feature is ready for deployment pending your final review.

**Comment by ewandennis** [comment]
:shipit:

**Pull request merged**